code/datums/diseases/anxiety.dm

```diff
@@ -36,6 +36,17 @@
 			if(prob(2))
 				affected_mob.visible_message("<span class='danger'>[affected_mob] coughs up butterflies!</span>", \
 													"<span class='userdanger'>You cough up butterflies!</span>")
-				new /mob/living/simple_animal/butterfly(affected_mob.loc)
-				new /mob/living/simple_animal/butterfly(affected_mob.loc)
+				//Yogs start -- Puts a maximum on the number of butterflies that can be spawned via this disease
+				var/mob/living/simple_animal/butterfly/Alpha = new(affected_mob.loc)
+				var/mob/living/simple_animal/butterfly/Beta = new(affected_mob.loc)
+				butterfly_count += list(Alpha,Beta)
+				if(butterfly_count.len > 250)
```

Comment:
This seems like an ultra clunky way to do it.
Could you not simply add a reference to this disease to the butterfly mobs, and then on death() have it remove itself from the list?